pkg/gce-pd-csi-driver/node.go

```diff
@@ -302,3 +317,35 @@ func (ns *GCENodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGe
 func (ns *GCENodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
 	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("NodeExpandVolume is not yet implemented"))
 }
+
+func (ns *GCENodeServer) GetVolumeLimits() (int64, error) {
+	// Default value 16 for memory lower than cutoff
+	volumeLimits := volumeLimit16
+
+	// Machine-type format: n1-type-CPUS or custom-CPUS-RAM
+	machineType := ns.MetadataService.GetMachineType()
+	if strings.HasPrefix(machineType, "n1-") {
+		// Predefined types with memory higher than cutoff
+		volumeLimits = volumeLimit128
+	} else if machineType == "f1-micro" || machineType == "g1-small" {
+		// Predefined types with memory lower than cutoff
+		return volumeLimits, nil
+	} else if strings.HasPrefix(machineType, "custom-") {
+		// Custom machine types
+		splits := strings.Split(machineType, "-")
+		if len(splits) != customTypeParts {
+			return volumeLimits, fmt.Errorf("Unexpected machine type: %s", machineType)
+		}
+		memStr := splits[customTypeMem]
+		mem, err := strconv.Atoi(memStr)
+
+		if (err == nil) && (mem >= memCutOff) {
+			volumeLimits = volumeLimit128
```

Comment:
Ok I got some extra clarification here.  Actually all custom machine types are 128.  There's no memory cutoff.

Reply:
ok, rules and test updated